salt/utils/event.py

```diff
@@ -401,9 +422,10 @@ def _get_event(self, wait, tag, match_func=None):
                 if socks.get(self.sub) != zmq.POLLIN:
                     continue
 
-                # Please do not use non-blocking mode here. Reliability is
-                # more important than pure speed on the event bus.
-                ret = self.get_event_block()
+                if no_block is True:
```

Comment:
Ah you're right, it must be when I tried - the while spun. (This is not my cod, just so you know)
I think the API here is that wait=0 means block infinite. Maybe my example above just needs modifying so it actually works if `get_event_noblock` throws (it was a gist of an idea). Otherwise can probably just do without it.